templates/_infrastructure_port_proxy.rb

```diff
@@ -0,0 +1,3 @@
+  ProxyPass / https://<%= @servername %>:9070
+  ProxyPassReverse / https://<%= @servername %>:9070
+  SSLProxyEngine On
```

Comment:
From some testing, this aspect of it will also require the following:

```
  ProxyPreserveHost on
  RequestHeader set X_FORWARDED_PROTO 'https'
  SSLProxyEngine On
  SSLProxyCACertificateFile "/var/lib/puppet/ssl/certs/ca.pem"
  SSLProxyMachineCertificateFile "/vagrant/foreman-installer/certs-bundle.pem"
```

This means we have to somewhat couple the 443 proxy port to the infrastructure port by making it aware of the CA that the infrastructure port is using (configured via  SSLProxyCACertificateFile). The other aspect is that we will have to generate a bundled key and certificate from the infrastructure CA and deploy it for the proxy to use (configured via SSLProxyMachineCertificateFile).